### PR TITLE
Cooldown – Optimize asset bundling and loading

### PR DESCRIPTION
**Pull request by ismaeldcom**
#### Changes proposed in this Pull Request
Right now we load most of the assets by creating a JS React file for SVG images or importing them from the `/assets` folder, which result in base64 data URI. The issue with this approach is that some images get bundled multiple times, because we have multiple entries configured.

In addition, we also load some of those images from PHP and CSS files, so we can't just generate the new files on `dist/` as some of them will be duplicated.

This PR includes the following changes:
- Update webpack config to inline `svg/png` files by default. And generate a URL to files in `assets/` folders, for those imported with the query `?assets`. Using the same pattern, we already use for chunks and load assets from PHP, `[file]?ver=[hash]` as cache busting
- Add a new alias for `assets` to easily import them.
- Replace every JS SVG component with SVG files.
- Adjust SCSS rules replacing `svg` selectors with `img`.
- Remove unused assets, duplicated ones, and those with a Gridicons version.
- Optimize every SVG and PNG files.
- Update jest config to handles assets imported with query, and update every affected snapshot.

#### Testing instructions
As we don't have visual regression tests, and this PR affect many different components, the best way to test it is to visually check that there is no difference between `develop` and this branch. Checking all the images from WCPay (Overview, Settings, Transactions, etc) load correctly, without any markup difference.

>**Note**: Some payment method icons will be slighly different, because I unified them to avoid having multiple versions of the same.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**
- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : QA Testing Not Applicable
- [x] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by htdat** [comment]
👋 @ismaeldcom, this is your cooldown issue, or it's a part of your team Moltres's sprint? 
If none of them is correct, I think you may create an issue for this PR to avoid getting it lost (as it's a nice idea and observation). 

FYI. I am currently doing triage as the Platform Porter PcreKM-yM-p2. I want to know the info above to categorize the issue/PR properly.

**Comment by ismaeldcom** [comment]
👋 Yes! This is a cooldown issue 🙂 Should we use a label or something to make the triage easier? Maybe just adding cooldown to the title 🤔

**Comment by htdat** [comment]
> Should we use a label or something to make the triage easier?

I don't think we need it as we do not have many issues/PRs like this. More infrequently used labels just make everyone more confused :D 

> Maybe just adding cooldown to the title.

Yes, or writing it somewhere in the PR description would be enough.

_Update: I've had another look, it seems that this issue has been in the `Current Focus` pipeline of Moltres' ZenHub board. That's also enough for me. I would have looked at this before._

**Comment by jbordonado** [comment]
How did you optimize the SVG files? Have you used something like https://github.com/svg/svgo?

Maybe we could have a pre-commit hook for this, to avoid any unoptimized assets

**Comment by ismaeldcom** [comment]
>How did you optimize the SVG files? Have you used something like https://github.com/svg/svgo?

I've used [SVGOMG](https://jakearchibald.github.io/svgomg/) to manually optimize each file.

>Maybe we could have a pre-commit hook for this, to avoid any unoptimized assets

I thought about adding webpack ImageMinimizerWebpackPlugin, but for SVG files it's better to do it manually to adjust the precision. Although using it with a safe configuration could be a good option, at least better than no optimization at all 🙂

File: client/components/payment-delete-illustration/test/index.tsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ import { render } from '@testing-library/react';
  * Internal dependencies
  */
 import PaymentDeleteIllustration from '..';
-import WooCardIcon from '../../../gateway-icons/woo-card';
+import WooCardIcon from 'assets/images/payment-methods/woo.svg?asset';
```
Comment (oaratovskyi) [inline client/components/payment-delete-illustration/test/index.tsx]: I wonder if it should be `woo-card.svg`?

**Review by oaratovskyi — APPROVED** [review]
Thanks, @ismaeldcom for this huuuge amount of work. I see the difference in our build size and it's dramatic, great work 🔥 

The only thing that I am curious about is the next change when the logo became bigger - is there any reason behind it?
image

Anyway, I've tested the following pages and haven't noticed any differences:
- Connect account
- Overview
- Deposit list
- Deposit details
- Transaction list
- Transaction details
- Dispute list
- Dispute details
- Settings
- Multi-Currency

I've also tested regular checkout as a shopper. I saw that there are changes in WooPay, but I haven't tested it, not sure how.

Pre-approving 🟢 
Let me know if you want me to test more pages and looking forward to your answer on the bigger logo!

**Comment by htdat** [comment]
👋 Hi all, this seems a low-risk PR but I see it's touching many files. I would suggest merging it after the next release of WCPay 5.6 (Mar 15) or after the cut-off commit for WCPay 5.6 freeze RC (Mar 12). Then it can go through all three testing cycles for WCPay 5.7.

File: client/components/payment-delete-illustration/test/index.tsx (hunk 1 above)
Comment (ismaeldcom) [reply client/components/payment-delete-illustration/test/index.tsx]: Nice catch! Addressed in 114c334

**Comment by ismaeldcom** [comment]
Thanks @oaratovskyi for reviewing this large PR! I think you've tested pretty much every page where images were used. If you're curious about, you can test WooPay by enabling platform checkout in **WCPay Dev**:

image

>The only thing that I am curious about is the next change when the logo became bigger - is there any reason behind it?

That's expected, as part of the optimization, I get rid of `width` and `height` definitions in SVG files, and kept only `viewbox`. That might make some images looks bigger or smaller. But in the end we're applying a fixed size later in JS/CSS.

>I would suggest merging it after the next release of WCPay 5.6 (Mar 15) or after the cut-off commit for WCPay 5.6 freeze RC (Mar 12). Then it can go through all three testing cycles for WCPay 5.7.

💯  agree @htdat! I also prefer to merge cooldown PRs at the beginning of a release to have plenty of testing cycles. Just in case 🙂

**Comment by oaratovskyi** [comment]
> I also prefer to merge cooldown PRs at the beginning of a release to have plenty of testing cycles

100% agree

**Comment by oaratovskyi** [comment]
@ismaeldcom there is also a conflict in `eps.svg`, I removed the comment in my [WCPay Things To Do PR](https://github.com/Automattic/woocommerce-payments/pull/5569/files), I shouldn't have done it :harold

**Comment by ismaeldcom** [comment]
>there is also a conflict in eps.svg, I removed the comment in my [WCPay Things To Do PR](https://github.com/Automattic/woocommerce-payments/pull/5569/files)

Yes, I noticed it while reviewing your PR 🙂  No problem, I'll fix it before merging, in case more conflicts arise.

**Comment by oaratovskyi** [comment]
Checked WooPay setting and checkout page, LGTM!

**Pull request merged**